presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java

```diff
@@ -62,6 +63,21 @@ private OperatorAssertion()
 
     public static List<Page> toPages(Operator operator, Iterator<Page> input)
     {
+        try {
+            return toPagesUnsafe(operator, input);
+        }
+        finally {
+            try {
+                operator.close();
```

Comment:
@haozhun  what do you mean by "migrate to the new method"? New method is with `DriverContext` and `OperatorFactory` instead of `Operator`?

Indeed it sounds better then current `toPages`. But maybe let's do this lazily as separate issue? I could revert changes in `toPages` and add this "new method" (with `OperatorFactory` argument) and use it in `TestHashAggregationOperator` wherever I can and as you stated use try-with-resrouces in `testMultiplePartialFlushes`. 

I don't want to do a large scale tests refactor so that this PR (and PRs that depend on this one) won't grow in size (as it happened few times in the past).